src/components/Characters/Characters.js

```diff
@@ -1,54 +1,201 @@
-      console.log('url: ', url)
-      console.log('info: ', info)
-      console.log('results', results)
-      console.log('search: ', search)
-    }, [info, url,search, results])
-  
-    useEffect(()=>{                     //combined url + result in input for search filter purpose
-        fetch(`${url}${search}`, {
-          method: 'GET'
-        }).then((response) => {
-          return response.json()
-        }).then((result) => {
-            setInfo(result.info)
-            setResults(result.results)
-        }).catch((error) => {
-            console.log(error)
-        })
-    }, [search])                        //Serach as a main dependency for this 
-  
-    return (
-        <div className="all-wrapper">
-            <div>
-                <input onChange={(e) => {
-                setSearch(e.target.value)
+/* eslint-disable no-shadow */
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/jsx-filename-extension */
+/* eslint-disable react/button-has-type */
+import React, { useState, useEffect, useReducer } from 'react';
+import Character from './Character';
+import './Characters.css';
+import FilterIcon from '../../content/icons/filter.png';
+import ArrowLeft from '../../content/icons/arrow-left.png';
+import ArrowRight from '../../content/icons/arrow-right.png';
+import Facebook from '../../content/icons/facebook.png';
+import LinkedIn from '../../content/icons/linkedin.png';
+import Github from '../../content/icons/github.png';
+import Filter from '../Filter/Filter';
+
+const initialState = {
+  info: {},
+  results: [],
+  search: '',
+  page: '1',
+  filter: false,
+  error: false,
+  url: 'https://rickandmortyapi.com/api/character/',
+};
+
+const ACTIONS = {
+  INFO: 'info',
+  RESULTS: 'results',
+  SEARCH: 'search',
+  PAGE: 'page',
+  FILTER: 'filter',
+  ERROR: 'error',
+  URL: 'url',
+};
+
+function reducer(state, action) {
+  switch (action.type) {
+    case ACTIONS.INFO:
+      return {
+        ...state,
+        info: action.payload,
+      };
+    case ACTIONS.RESULTS:
+      return {
+        ...state,
+        results: action.payload,
+      };
+    case ACTIONS.SEARCH:
+      return {
+        ...state,
+        search: action.payload,
+      };
+    case ACTIONS.PAGE:
+      return {
+        ...state,
+        page: action.payload,
+      };
+    case ACTIONS.FILTER:
+      return {
+        ...state,
+        filter: action.payload,
+      };
+    case ACTIONS.ERROR:
+      return {
+        ...state,
+        error: action.payload,
+      };
+    case ACTIONS.URL:
+      return {
+        ...state,
+        url: action.payload,
+      };
+    default:
+      return state;
+  }
+}
+
+function Characters() {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const {
+    info, results, search, page, filter, error, url,
+  } = state;
+  const [addition, setAddition] = useState('');
+
+  const getDataValue = (value) => {
```

Comment:
which data? what value? let's name it less general

Reply:
Sure. You 100% right.